pallets/gas/src/tests.rs

```diff
@@ -46,6 +46,41 @@ fn simple_value_tree() {
     });
 }
 
+#[test]
+fn value_tree_with_all_kinds_of_nodes() {
+    new_test_ext().execute_with(|| {
+        let total_supply = 1000;
+        let cut_value = 500;
+        let specified_value = total_supply - cut_value;
+        let (root, cut, specified, unspecfied) = (
+            H256::random(),
+            H256::random(),
+            H256::random(),
+            H256::random(),
+        );
+
+        // create nodes
+        {
+            assert_ok!(Gas::create(ALICE.into_origin(), root, total_supply));
+            assert_ok!(Gas::cut(root, cut, cut_value));
+            assert_ok!(Gas::split_with_value(root, specified, specified_value));
+            assert_ok!(Gas::split(root, unspecfied));
+        }
+
+        assert_eq!(Gas::total_supply(), total_supply);
+
+        // consume nodes
+        {
+            assert_ok!(Gas::consume(unspecfied));
+            assert_ok!(Gas::consume(specified));
+            assert_ok!(Gas::consume(root));
```

Comment:
please add here that the returned result `is_some` and that root was deleted, so `GasTree` has only one node - the reserved one.